### PR TITLE
Fixing the codebase, new travis ( dreamchecker as static analyst ) and for 3rd party webmap ( dmm-tools )

### PR DESCRIPTION
**Pull request by TorinoFermic**
## About The Pull Request

These changes are big, but they are required make possible of using travis/dreamchecker like others popular repos. Because BYOND Compiler can't be trusted AT ALL

I am fixing the code that has errors or warnings from dreamchecker.ext of https://github.com/SpaceManiac/SpacemanDMM for future travis automation. Also changes random spawners to be invisible for the minimap renderer

- [x] ~~Port inventory.dm and their dependencies from Baystation12~~ Instead I used correct variables, porting BS12 isn't a easy feat
- [x] Added SpacemanDMM.toml for VS Code to enable the use dreamchecker and reveal more mistakes

FIX: #4219

Current progress : 0 diagnostics

Ignore
code/modules/shield_generators/shield.dm It was fixed in other pr.

FIXES
code/game/machinery/machinery.dm : bad var in a loop
code/modules/power/sensors/sensor_monitoring.dm Fixes the code flow
code/modules/mob/living/silicon/robot/inventory.dm De-mangle the logic of activating modules
code/modules/mob/living/living_defense.dm Why this line
code/modules/mob/living/carbon/human/human_defense.dm Somehow, this line has compiled and stayed cursed until dmm-tools finds it
code/modules/mining/machine_processing.dm avoids bad overload logic
code/modules/hydroponics/grown.dm Fixes broken loop
code/modules/economy/cash_register.dm Incorrect argv type
code/modules/admin/bluespacetech.dm QoL for BST
code/game/objects/random/random.dm makes all map spawner icons invisible for minimap renderer
code/game/objects/items/devices/taperecorder.dm Fixes the tape recorder ?
code/game/machinery/suit_storage_unit.dm Fixes defines due having duplicate defines
code/game/machinery/portable_turret.dm Fixes mistakes of another coder
code/datums/wires/vending.dm Defines at correct place
code/__DEFINES/spaceman_dmm.dm Adds required defines for dmm-tools

### CODE REVIEW

File: code/game/objects/items/weapons/storage/storage.dm [hunk 1]
```diff
@@ -55,7 +55,7 @@
 		S.close(clientMob)
 
 /obj/item/weapon/storage/proc/setupItemBackground(var/HUD_element/itemBackground, var/atom/item, var/itemCount)
-	itemBackground.setClickProc(.itemBackgroundClick)
+	itemBackground.setClickProc(UNLINT(.itemBackgroundClick))
```
Comment (spookydonut) [inline code/game/objects/items/weapons/storage/storage.dm:58]: not a great way to deal with this but its fine as long as someone comes back later to refactor this.

File: code/modules/admin/verbs/mapping.dm [hunk 2]
```diff
@@ -90,7 +88,7 @@ var/intercom_range_display_status = 0
 			if(!(locate(/obj/structure/grille,T)))
 				var/window_check = 0
 				for(var/obj/structure/window/W in T)
-					if (W.dir == turn(C1.dir,180) || W.dir in list(5,6,9,10) )
+					if (W.dir == turn(C1.dir,180) || (W.dir in list(5,6,9,10)))
```
Comment (spookydonut) [inline code/modules/admin/verbs/mapping.dm:91]: change these into the direction defines `(NORTHEAST, SOUTHEAST, NORTHWEST, SOUTHWEST)`

File: code/modules/mining/machine_processing.dm [hunk 3]
```diff
@@ -224,11 +224,11 @@
 					continue
 
 				for(var/i=0,i<can_make,i++)
-					ores_stored[metal]--
```
Comment (spookydonut) [inline code/modules/mining/machine_processing.dm:227]: -- is fine, don't make this a typed list if only the key is a type.

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 4]
```diff
@@ -152,7 +152,7 @@
 						ears.loc = src.loc
 						ears = null
 						for(var/possible_phrase in speak)
-							if(copytext(possible_phrase,1,2) == get_prefix_key(/decl/prefix/radio_channel_selection) && copytext(possible_phrase,2,3) in department_radio_keys)
+							if(copytext(possible_phrase,1,2) == get_prefix_key(/decl/prefix/radio_channel_selection) && (copytext(possible_phrase,2,3) in department_radio_keys))
```
Comment (spookydonut) [inline code/modules/mob/living/simple_animal/parrot.dm:155]: copytext should probably be replaced with copytext_char

File: code/modules/modular_computers/file_system/programs/command/comm.dm [hunk 5]
```diff
@@ -301,19 +301,19 @@ var/last_message_id = 0
 	if(!frequency) return
 
 	var/datum/signal/status_signal = new
-	status_signal.source = src
+	status_signal.source = UNLINT(src)
```
Comment (spookydonut) [inline code/modules/modular_computers/file_system/programs/command/comm.dm:304]: src is meaningless in global procs

File: code/modules/reagents/metabolism.dm [hunk 6]
```diff
@@ -198,7 +198,7 @@
 			addiction_list.Remove(R)
 			continue
 
-		addiction_list[R]++
+		addiction_list[R] += 1
```
Comment (spookydonut) [inline code/modules/reagents/metabolism.dm:201]: same

File: code/modules/modular_computers/file_system/programs/command/comm.dm (hunk 5 above)
Comment (TorinoFermic) [reply code/modules/modular_computers/file_system/programs/command/comm.dm:304]: Not sure what you mean, would the code down frequency.post_signal() needs it ?

Comment (spookydonut) [reply code/modules/modular_computers/file_system/programs/command/comm.dm]: src isn't a thing in global procs, you might mean usr?

Comment (TorinoFermic) [reply code/modules/modular_computers/file_system/programs/command/comm.dm]: I see, I checked agaisn't BayStation12 repo, That's true, I don't find any src there. Question resolved

**Comment by TorinoFermic** [comment]
Well, since ACCount asking which issues it fixes. I can only bring one that's visible to everyone because they noticed but under the code, dreamchecker can find the hidden broken stuff.

**Comment by TorinoFermic** [comment]
Leaving it open for others to comment my changes and offer suggestions.

**Pull request merged**